src/components/Combobox/Combobox.tsx

```diff
@@ -35,6 +35,11 @@ export type ComboboxProps<T> = {
    * values to the user) */
   itemToString?: (item: T) => string;
 
+  /**
+   * A function used to group the Menu Items.
+   */
+  itemToGroup?: (item: T) => string | undefined;
```

Comment:
I think that if you decide you need to group items, every item should have a group. Even the ones that you don't have to "group" should belong somewhere. For example, you could have a group named "Other" for all the ones that don't fall under any other group


```suggestion
  itemToGroup?: (item: T) => string;
```